drivers/linstorvhdutil.py

```diff
@@ -83,9 +80,16 @@ def log_successful_call(target_host, device_path, vdi_uuid, remote_method, respo
         priority=util.LOG_DEBUG
     )
 
-def log_failed_call(target_host, next_target, device_path, vdi_uuid, remote_method, e):
+def log_failed_call(target_host, next_target, device_path, vdi_uuid, remote_method, e=None):
     util.SMlog(
-        'Failed to call method on {} for device {} ({}): {}. Trying accessing on {}... (cause: {})'.format(target_host, device_path, vdi_uuid, remote_method, next_target, e),
+        'Failed to call method on {} for device {} ({}): {}. Trying accessing on {}... (cause: {})'.format(
+            target_host,
+            device_path,
+            vdi_uuid,
+            remote_method,
+            next_target,
+            e if e else "no error provided"
```

Comment:
I think you can give a string like "no primary" to `log_failed_call` and remove the default value of `e` for this specific case.

Reply:
fixed in [614cda9](https://github.com/xcp-ng/sm/pull/93/commits/614cda9525ac5d75c1897c5c107780978037b98e)